The install_name of a library can be compared to the ELF soname, setting
it to just the library name (libx.1.dylib) is not sufficient on Mach-O,
as it uses full paths.  Most apps have no clue about that, so they don't
do something like http://www.cmake.org/pipermail/cmake/2006-June/009758.html

Provide a sane default that will allow merging most cmake-based
packages.  If this location is not correct, Portage's QA checks will
catch that.

--- Source/cmTarget.cxx
+++ Source/cmTarget.cxx
@@ -974,7 +974,7 @@
                        this->Makefile->IsOn("MINGW"));
 
   // Setup default property values.
-  this->SetPropertyDefault("INSTALL_NAME_DIR", "");
+  this->SetPropertyDefault("INSTALL_NAME_DIR", "${CMAKE_INSTALL_PREFIX}/lib");
   this->SetPropertyDefault("INSTALL_RPATH", "");
   this->SetPropertyDefault("INSTALL_RPATH_USE_LINK_PATH", "OFF");
   this->SetPropertyDefault("SKIP_BUILD_RPATH", "OFF");
